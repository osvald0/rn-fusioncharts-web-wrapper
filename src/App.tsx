import React from 'react';
import { StyleSheet, View, ViewStyle } from 'react-native';
import FusionCharts from './components/FusionCharts';

type Props = {
  chartConfig: unknown;
  containerStyle?: ViewStyle;
};

const App = (props: Props): JSX.Element => {
  const { chartConfig, containerStyle } = props;
  return (
    <View style={{ ...styles.containerStyle, ...containerStyle }}>
      <FusionCharts chartConfig={chartConfig} />
    </View>
  );
};

const styles = StyleSheet.create({
  containerStyle: {
    width: '100%',
    height: '100%',
  },
});

export default App;
