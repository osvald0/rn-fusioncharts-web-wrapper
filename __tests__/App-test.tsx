import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import App from '../src/App';

jest.mock('react-native-webview', () => 'WebView');

const FUSIONCHARTS_DATA = {
  type: 'column2d',
  renderAt: 'chart-container',
  width: '600',
  height: '400',
  dataFormat: 'json',
  dataSource: {
    chart: {
      caption: 'Countries With Most Oil Reserves [2017-18]',
      subcaption: 'In MMbbl = One Million barrels',
    },
    data: [
      { label: 'Venezuela', value: '290000' },
      { label: 'Saudi', value: '260000' },
      { label: 'Canada', value: '180000' },
      { label: 'Iran', value: '140000' },
      { label: 'Russia', value: '115000' },
      { label: 'UAE', value: '100000' },
      { label: 'US', value: '30000' },
      { label: 'China', value: '30000' },
    ],
  },
};

describe('App', () => {
  it('Renders correctly', () => {
    const app = renderer.create(<App chartConfig={FUSIONCHARTS_DATA} />).toJSON();
    expect(app).toMatchSnapshot();
  });
});
