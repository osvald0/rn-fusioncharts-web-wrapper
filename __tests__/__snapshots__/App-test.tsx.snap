// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders correctly 1`] = `
<View
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <WebView
    originWhitelist={
      [
        "*",
      ]
    }
    source={
      {
        "html": "
    <!doctype html>
    <html>
    <head>
        <script src="https://cdn.fusioncharts.com/fusioncharts/latest/fusioncharts.js"></script>
    </head>
    <body>
        <div id="chart-container"></div>
        <script>
            FusionCharts.ready(function () {
                var chart = new FusionCharts({"type":"column2d","renderAt":"chart-container","width":"600","height":"400","dataFormat":"json","dataSource":{"chart":{"caption":"Countries With Most Oil Reserves [2017-18]","subcaption":"In MMbbl = One Million barrels"},"data":[{"label":"Venezuela","value":"290000"},{"label":"Saudi","value":"260000"},{"label":"Canada","value":"180000"},{"label":"Iran","value":"140000"},{"label":"Russia","value":"115000"},{"label":"UAE","value":"100000"},{"label":"US","value":"30000"},{"label":"China","value":"30000"}]}}).render();
            });
        </script>
    </body>
    </html>
",
      }
    }
  />
</View>
`;
